tests/concurrent_test.go

```diff
@@ -71,3 +72,73 @@ func TestProxy_Concurrency(t *testing.T) {
 	}
 	wg.Wait()
 }
+
+func TestAgent_MultipleConn(t *testing.T) {
+	testcases := []struct {
+		name                string
+		proxyServerFunction func() (proxy, func(), error)
+		clientFunction      func(string, string) (*http.Client, error)
+	}{
+		{
+			name:                "grpc",
+			proxyServerFunction: runGRPCProxyServer,
+			clientFunction:      createGrpcTunnelClient,
+		},
+		{
+			name:                "http-connect",
+			proxyServerFunction: runHTTPConnProxyServer,
+			clientFunction:      createHTTPConnectClient,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			echoServer := newEchoServer("hello")
+			echoServer.wchan = make(chan struct{})
+			server := httptest.NewServer(echoServer)
+			defer server.Close()
+
+			stopCh := make(chan struct{})
+			stopCh2 := make(chan struct{})
+
+			proxy, cleanup, err := tc.proxyServerFunction()
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer cleanup()
+
+			runAgentWithID("multipleAgentConn", proxy.agent, stopCh)
+			defer close(stopCh)
+
+			// Wait for agent to register on proxy server
+			wait.Poll(100*time.Millisecond, 5*time.Second, func() (bool, error) {
+				ready, _ := proxy.server.Readiness.Ready()
+				return ready, nil
+			})
+
+			// run test client
+			c, err := tc.clientFunction(proxy.front, server.URL)
+
+			fcnStopCh := make(chan struct{})
+
+			go func() {
+				_, err := clientRequest(c, server.URL)
+				if err != nil {
+					t.Errorf("expected no error on proxy request, got %v", err)
+				}
+				close(fcnStopCh)
+			}()
+
+			// Running an agent with the same ID simulates a second connection from the same agent.
+			// This scenario is often hit when running a HA proxy server
```

Comment:
How about

"Running an agent with the same ID simulates a second connection from the same agent. This simulates the scenario where a proxy agent established connections with HA proxy server and creates multiple connections with the same proxy server."